conans/client/loader.py

```diff
@@ -20,12 +20,11 @@
 
 
 class ProcessedProfile(object):
-    def __init__(self, settings=None, profile=None, create_reference=None):
-        settings = settings or Settings()
-        profile = profile or Profile()
-        assert isinstance(settings, Settings)
-        # assert package_settings is None or isinstance(package_settings, dict)
-        self._settings = settings
+    def __init__(self, profile=None, create_reference=None):
+        if profile is None:  # FIXME: Only for testing interface
+            profile = Profile()
+            profile.processed_settings = Settings()
```

Comment:
If it is just for the tests, make this argument compulsory. The caller from tests will be able to add the required two lines, create a factory method, inherit the class...

Reply:
Didn't want to add those many changes in the test, to keep PR shorter, but yes, lets do it.